src/sentry/ownership/grammar.py

```diff
@@ -86,6 +90,35 @@ def load(cls, data):
             data['pattern'],
         )
 
+    def test(self, data):
+        return getattr(self, 'test_%s' % self.type)(data)
+
+    def test_url(self, data):
+        try:
+            url = data['sentry.interfaces.Http']['url']
+        except KeyError:
+            return False
+        return fnmatch(url, self.pattern)
+
+    def test_path(self, data):
+        for frame in _iter_frames(data):
+            try:
+                filename = frame['filename']
+            except KeyError:
+                try:
+                    filename = frame['abs_path']
+                except KeyError:
+                    continue
+
+            # fnmatch keeps it's own internal cache, so
+            # there isn't any optimization we can do here
+            # by using fnmatch.translate before and compiling
+            # our own regex.
+            if fnmatch(filename, self.pattern):
+                return True
+
+        return False
```

Comment:
Can't this just be

```py
return fnmatch(filename, self.pattern)
```

Reply:
No, because if we don't match one frame, we need to continue checking through all the frames, so we don't want to exit early.

Reply:
Ah my mistake, for some reason I read the last `return False` as being under the loop, which of course doesn't make sense. yup